src/framework/execution/__tests__/logger.test.ts

```diff
@@ -431,7 +432,102 @@ describe('createErrorEventDescription', () => {
       'testing',
     );
     expect(description).toEqual(
-      `testing (errorCode=${error.code}, errorId=${errorId}, reason=soba)`,
+      `testing (errorCode="${error.code}", errorId="${errorId}", reason="soba")`,
     );
   });
 });
+
+describe('#publishEvent', () => {
```

Comment:
I had hoped to keep from having integrations publish random events to the job log because I had noticed that when that feature was available, there was no consistency in the logging across integrations, and the information presented to users was haphazard. I do think we need to communicate more to users than we are currently able to communicate, but I think it to be particularly errors that are non-terminal during a step, and which should not cause a Pager Duty event, so I had hoped that we could use` context.logger.warn({ err, ... }, "Optional message here")`, and then re-use types found in https://github.com/JupiterOne/integration-sdk/blob/master/src/framework/execution/error.ts. This would allow us to develop consistent presentation of events as well as dashboards for debugging.

So it would come out as:

```ts
try {
  ... some API call ...
} catch(err) {
  // I can handle this, but want to note it
  logger.warn({
    err: new IntegrationProviderAPIError({
      cause: err,
      endpoint: "https://provider.com/api/resource",
      status: response.status,
      statusText: response.statusText,
    })
  });
}
```